doc/check-options.md

```diff
@@ -0,0 +1,215 @@
+# Check Options
+
+## Table of Contents
+
+- [How Checks Work](#how-checks-work)
+- [Options](#options)
+  - [aria-allowed-role](#aria-allowed-role)
+  - [aria-required-children](#aria-required-children)
+  - [aria-roledescription](#aria-roledescription)
+  - [color-contrast](#color-contrast)
+  - [page-has-heading-one](#page-has-heading-one)
+  - [page-has-main](#page-has-main)
+  - [page-no-duplicate-banner](#page-no-duplicate-banner)
+  - [page-no-duplicate-contentinfo](#page-no-duplicate-contentinfo)
+  - [page-no-duplicate-main](#page-no-duplicate-main)
+  - [duplicate-img-label](#duplicate-img-label)
+  - [label-content-name-mismatch](#label-content-name-mismatch)
+  - [has-lang](#has-lang)
+  - [valid-lang](#valid-lang)
+  - [frame-tested](#frame-tested)
+  - [no-autoplay-audio](#no-autoplay-audio)
+  - [css-orientation-lock](#css-orientation-lock)
+  - [meta-viewport-large](#meta-viewport-large)
+  - [meta-viewport](#meta-viewport)
+  - [header-present](#header-present)
+  - [landmark](#landmark)
+  - [p-as-heading](#p-as-heading)
+  - [avoid-inline-spacing](#avoid-inline-spacing)
+  - [scope-value](#scope-value)
+
+## How Checks Work
+
+[Rules in axe-core](../lib/rules) are made up of individual checks that dictate how the rule works. Each check is typically designed to look for a specific requirement and report back its findings to the rule.
+
+For example, the rule [image-alt](../lib/rules/image-alt.json) uses the checks `has-alt`, `aria-label`, `aria-labelledby`, and `non-empty-title` to determine if the image has an accessible name from an `alt`, `aria-label`, `aria-labelledby`, or `title` attribute (respectively).
+
+Many checks allow you to change how they work through `options` properties. These options can be found in the [checks metadata file](../lib/checks).
+
+For example, the check [has-lang](../lib/checks/language/has-lang.json) takes an `attributes` option which dictates which attributes to check for a lang value.
+
+To customize a checks options, you can use [`axe.configure`](./API.md#api-name-axeconfigure) to configure the check and modify the options as desired.
+
+```js
+// configure has-lang check to look at the `hreflang` attribute as well
+axe.configure({
+	checks: [
+		{
+			id: 'has-lang',
+			options: {
+				attributes: ['lang', 'xml:lang', 'hreflang']
+			}
+		}
+	]
+});
+```
```

Comment:
something's wonky with the rendered codelisting formatting here:

![image](https://user-images.githubusercontent.com/4044039/88216012-8200ba80-cc2a-11ea-8a41-abaa5fe04caa.png)

Reply:
Aside from it being hard to read (due to the tabs), I don't see anything "wonky" here.

Reply:
It looks like a linter gone wild thing. The code itself looks fine. Github rendering, not so much.

Reply:
I'm not sure what you're suggesting here. Is there anything that should be "fixed"?

Reply:
@stephenmathieson: not if the 2 spaces = indentation/prettier stuff is working, no. It's up to Github to render it properly.

Reply:
I do not understand.